libfreerdp/cache/bitmap.c

```diff
@@ -278,6 +282,108 @@ void bitmap_cache_register_callbacks(rdpUpdate* update)
 	}
 }
 
+int bitmap_cache_load_persistent(rdpBitmapCache* bitmapCache)
+{
+	int index;
+	int count;
+	int status;
+	rdpPersistentCache* persistent;
+	PERSISTENT_CACHE_ENTRY cacheEntry;
+	rdpContext* context = bitmapCache->context;
+	rdpSettings* settings = context->settings;
+
+	if (!settings->BitmapCachePersistEnabled)
+		return 0;
+
+	if (!settings->BitmapCachePersistFile)
+		return 0;
+
+	persistent = persistent_cache_new();
+
+	if (!persistent)
+		return -1;
+
+	status = persistent_cache_open(persistent, settings->BitmapCachePersistFile, FALSE, -1);
+
+	if (status < 1)
+		goto error;
+
+	count = persistent_cache_get_count(persistent);
+
+	for (index = 0; index < count; index++)
+	{
+		if (persistent_cache_read_entry(persistent, &cacheEntry) < 1)
+			continue;
+	}
+
+	persistent_cache_free(persistent);
```

Comment:
Maybe use a return variable, so the `success` and `error` case here can be eliminated?